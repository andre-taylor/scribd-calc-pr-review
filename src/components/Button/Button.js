import React from "react"
import PropTypes from "prop-types"
import cx from "classnames"

import styles from "./Button.module.css"

const Button = ({ buttonValue }) => {
  const getClassName = buttonValue => {
    let cssName
    switch (buttonValue) {
      case "+":
        cssName = "add"
        break
      case "-":
        cssName = "subtract"
        break
      case "*":
        cssName = "multiple"
        break
      case "/":
        cssName = "divide"
        break
      case "=":
        cssName = "calculate"
        break
      case "C":
        cssName = "clearScreen"
        break
      default:
        cssName = `key${buttonValue}`
        break
    }
    return cssName
  }
  return (
    <button type="button" value={buttonValue} className={cx(styles.button, styles[getClassName(buttonValue)])}>
      {buttonValue}
    </button>
  )
}

export default Button

Button.propTypes = {
  buttonValue: PropTypes.string.isRequired
}
