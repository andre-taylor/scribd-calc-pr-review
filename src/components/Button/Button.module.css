.button {
  cursor: pointer;
  height: 60px;
  background-color: #fff;
  border-radius: 3px;
  border: 1px solid #c4c4c4;
  background-color: transparent;
  font-size: 2rem;
  color: #333;
  background-image: linear-gradient(to bottom, transparent, transparent 50%, rgba(0, 0, 0, 0.04));
  box-shadow: inset 0 0 0 1px rgba(255, 255, 255, 0.05), inset 0 1px 0 0 rgba(255, 255, 255, 0.45),
    inset 0 -1px 0 0 rgba(255, 255, 255, 0.15), 0 1px 0 0 rgba(255, 255, 255, 0.15);
  text-shadow: 0 1px rgba(255, 255, 255, 0.4);
}
.button:hover {
  background-color: #eaeaea;
}
.key0 {
  grid-column: span 2;
}
.multiple {
  line-height: 2.4;
}
.calculate {
  background-color: #2e86c0;
  border-color: #337cac;
  color: #fff;
  height: 100%;
  grid-row-start: 2;
  grid-row-end: 6;
  grid-column-start: 4;
  grid-column-end: 5;
}
.calculate:hover {
  background-color: #4e9ed4;
}
.clearScreen {
  background-color: #f0595f;
  border-color: #b0353a;
  color: #fff;
}
.clearScreen:hover {
  background-color: #f17377;
}
.selected {
  background-color: #c0c0c0;
  color: #f0f0f0;
}
