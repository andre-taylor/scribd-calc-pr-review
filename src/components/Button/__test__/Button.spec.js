import React from "react"
import renderer from "react-test-renderer"

import Button from "../Button"

describe("<Button />", () => {
  it("renders correctly with a '+' switch value", () => {
    const component = renderer.create(<Button buttonValue={"+"} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with a '-' switch value", () => {
    const component = renderer.create(<Button buttonValue={"-"} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with a '*' switch value", () => {
    const component = renderer.create(<Button buttonValue={"*"} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with a '/' switch value", () => {
    const component = renderer.create(<Button buttonValue={"/"} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with a '=' switch value", () => {
    const component = renderer.create(<Button buttonValue={"="} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with a 'C' switch value", () => {
    const component = renderer.create(<Button buttonValue={"C"} />)
    expect(component).toMatchSnapshot()
  })
  it("renders correctly with a default switch value", () => {
    const component = renderer.create(<Button buttonValue={"0"} />)
    expect(component).toMatchSnapshot()
  })
})
