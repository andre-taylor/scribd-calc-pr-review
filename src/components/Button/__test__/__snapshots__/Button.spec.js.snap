// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly with a '*' switch value 1`] = `
<button
  className="button multiple"
  type="button"
  value="*"
>
  *
</button>
`;

exports[`<Button /> renders correctly with a '+' switch value 1`] = `
<button
  className="button add"
  type="button"
  value="+"
>
  +
</button>
`;

exports[`<Button /> renders correctly with a '/' switch value 1`] = `
<button
  className="button divide"
  type="button"
  value="/"
>
  /
</button>
`;

exports[`<Button /> renders correctly with a '=' switch value 1`] = `
<button
  className="button calculate"
  type="button"
  value="="
>
  =
</button>
`;

exports[`<Button /> renders correctly with a '-' switch value 1`] = `
<button
  className="button subtract"
  type="button"
  value="-"
>
  -
</button>
`;

exports[`<Button /> renders correctly with a 'C' switch value 1`] = `
<button
  className="button clearScreen"
  type="button"
  value="C"
>
  C
</button>
`;

exports[`<Button /> renders correctly with a default switch value 1`] = `
<button
  className="button key0"
  type="button"
  value="0"
>
  0
</button>
`;
